### PR TITLE
[spec] Fix grammar for fractions

### PR DESCRIPTION
**Pull request by rossberg**
Fixes WebAssembly/design#1337.

### CODE REVIEW

File: document/core/text/values.rst [hunk 1]
```diff
@@ -105,11 +103,11 @@ Floating-Point
      p{:}\Tnum~\text{.}~q{:}\Tfrac~(\text{E}~|~\text{e})~{\pm}{:}\Tsign~e{:}\Tnum
        &\Rightarrow& (p+q)\cdot 10^{\pm e} \\
    \production{hexadecimal floating-point number} & \Thexfloat &::=&
-     \text{0x}~p{:}\Thexnum
+     \text{0x}~p{:}\Thexnum~\text{.}
```
Comment (binji) [inline document/core/text/values.rst:106]: The dot should be optional here too?

File: document/core/text/values.rst [hunk 2]
```diff
@@ -105,11 +103,11 @@ Floating-Point
...
        &\Rightarrow& p \\ &&|&
      \text{0x}~p{:}\Thexnum~\text{.}~q{:}\Thexfrac
        &\Rightarrow& p+q \\ &&|&
-     \text{0x}~p{:}\Thexnum~(\text{P}~|~\text{p})~{\pm}{:}\Tsign~e{:}\Tnum
+     \text{0x}~p{:}\Thexnum~\text{.}^?~(\text{P}~|~\text{p})~{\pm}{:}\Tsign~e{:}\Tnum
```
Comment (binji) [inline document/core/text/values.rst:110]: Same change for decimal floats above? (line 101)

File: document/core/text/values.rst (hunk 1 above)
Comment (rossberg) [reply document/core/text/values.rst]: Indeed, fixed.

File: document/core/text/values.rst (hunk 2 above)
Comment (rossberg) [reply document/core/text/values.rst:110]: Fixed.

**Pull request merged**